docs/apps/lnw/es2k/es2k-linux-networking-lag.md

```diff
@@ -0,0 +1,336 @@
+ip netns exec VM0  ifconfig <VF1 port> up
+
+ip netns add VM1
+ip link set <VF2 port> netns VM1
+ip netns exec VM1 ip addr add 99.0.0.2/24 dev <VF2 port>
+ip netns exec VM1 ifconfig <VF2 port> up
+```
+
+### Start OvS as a separate process
+
+Legacy OvS is used as a control plane for source MAC learning of overlay VM's. OvS should be started as a seperate process.
+
+```bash
+export RUN_OVS=/tmp
+rm -rf $RUN_OVS/etc/openvswitch
+rm -rf $RUN_OVS/var/run/openvswitch
+mkdir -p $RUN_OVS/etc/openvswitch/
+mkdir -p $RUN_OVS/var/run/openvswitch
+
+ovsdb-tool create $RUN_OVS/etc/openvswitch/conf.db \
+    /opt/p4/p4-cp-nws/share/openvswitch/vswitch.ovsschema
+
+ovsdb-server $RUN_OVS/etc/openvswitch/conf.db \
+    --remote=punix:$RUN_OVS/var/run/openvswitch/db.sock \
+    --remote=db:Open_vSwitch,Open_vSwitch,manager_options \
+    --pidfile=$RUN_OVS/var/run/openvswitch/ovsdb-server.pid \
+    --unixctl=$RUN_OVS/var/run/openvswitch/ovsdb-server.ctl \
+    --detach
+
+ovs-vswitchd --detach \
+    --pidfile=$RUN_OVS/var/run/openvswitch/ovs-vswitchd.pid \
+    --no-chdir unix:$RUN_OVS/var/run/openvswitch/db.sock \
+    --unixctl=$RUN_OVS/var/run/openvswitch/ovs-vswitchd.ctl \
+    --mlockall \
+    --log-file=/tmp/ovs-vswitchd.log
+
+alias ovs-vsctl="ovs-vsctl --db unix:$RUN_OVS/var/run/openvswitch/db.sock"
+ovs-vsctl set Open_vSwitch . other_config:n-revalidator-threads=1
+ovs-vsctl set Open_vSwitch . other_config:n-handler-threads=1
+
+ovs-vsctl  show
+```
+
+### Create VLAN representers
+
+For each VM that is spawned for overlay network we need to have a port representer.
+We create VLAN netdevs on top of the IPDF netdev which is assigned to VSI group 3 in step-2 mentioned above.
+
+```bash
+ip link add link <VSI 8> name vlan1 type vlan id 1
+ip link add link <VSI 8> name vlan2 type vlan id 2
+ifconfig vlan1 up
+ifconfig vlan2 up
+```
+
+Note: Here the assumption is, we have created 2 overlay VM's and creating 2 port representers for those VM's.
+Port representer should always be in the format: `lowercase string 'vlan'+'vlanID'`
+
+### Create integration bridge and add ports to the bridge
+
+Create OvS bridge, VxLAN tunnel and assign ports to the bridge.
+
+```bash
+ovs-vsctl add-br br-int
+ifconfig br-int up
+
+ovs-vsctl add-port br-int vlan1
+ovs-vsctl add-port br-int vlan2
+ifconfig vlan1 up
+ifconfig vlan2 up
+
+ovs-vsctl add-port br-int vxlan1 -- set interface vxlan1 type=vxlan \
+    options:local_ip=30.1.1.1 options:remote_ip=40.1.1.1 options:dst_port=4789
+```
+
+Note: Here we are creating VxLAN tunnel with VNI 0, you can create any VNI for tunneling.
+
+### Configure rules for overlay control packets
+
+Configure rules to send overlay control packets from a VM to its respective port representers.
+
+Below configuration assumes
+
+- Overlay VF1 has a VSI value 14
+- Overlay VF2 has a VSI value 15
+
+These VSI values can be checked with `/usr/bin/cli_client -q -c` command
+on IMC.  This command provides VSI ID, Vport ID, and corresponding MAC
```

Comment:
```suggestion
on IMC.  This command provides VSI ID, vPort ID, and corresponding MAC
```

Reply:
I have no problem with _Vport_. _vport_ is a term of art meaning "virtual port"; it doesn't require non-standard capitalization.

VSI and MAC are in all caps because they're initialisms; ID is in all caps because that's the standard way of abbreviating _identifier_.

It's not uncommon to capitalize a non-initialism in a list like this -- it's a visual cue that the word is an item in the list, not a conjunction like _and_ or a modifier like _corresponding_. _port ID_ would probably become _Port ID_ because _port_ is a word. _vport ID_ could be rendered as _Vport ID_ (treating it as a word) or _VPORT ID_ (treating it as an acronym, which it is).

Reply:
I am using Vport for now. Let me know if I need to change it.